js/ui/marker.js

```diff
@@ -81,7 +87,7 @@ Marker.prototype = {
 
     _update: function() {
         if (!this._map) return;
-        var pos = this._map.project(this._lngLat);
+        var pos = this._map.project(this._lngLat).add(this._offset);
```

Comment:
Minor remark: we can do `_add` instead of `add` here to avoid allocating a new point object.